docs/victoriametrics/data-ingestion: add OpenShift guide (#9790)
VictoriaMetrics/VictoriaMetrics: docs/victoriametrics/data-ingestion/OpenShift.md

> ---

Reviewer: We have these sections /integrations and /data-ingestion and they might be a bit confusing for users, and for us. Confusing because it is not clear where to put new docs:
* we configured RW into openshift, so it could be part of /data-ingestion
* but openshift is a product itself, so it could be part of /integration with this product

We should agree on the logic of using these two sections. I propose the following:
1. Everything that describes how to work with specific API or tech details of protocol should be part of /data-ingestion. It also means that https://docs.victoriametrics.com/victoriametrics/#how-to-import-time-series-data should be part of /data-ingestion page.
2. Everything that describes how to connect with other products should go to /integrations. To be simple, everything that requires updating .yaml files is kinda an integration.

@makasim @vrutkovs we should agree on this before we proceed. If you agree, than this doc should be part of /integrations page.

title: OpenShift
weight: 7
menu:
  docs:
    identifier: "openshift"
    parent: "data-ingestion"
    weight: 7
tags:
  - metrics
aliases:
  - /data-ingestion/openshift.html
  - /data-ingestion/OpenShift.html
  - /data-ingestion/openshift/index.html
  - /data-ingestion/openshift/
---

## OpenShift Container Platform (OCP)

OpenShift uses Prometheus as a core monitoring solution. It cannot be replaced without violating the OpenShift support policy. However, OpenShift can be configured to use VictoriaMetrics as a remote write target.

According to [remote write configuration in the OpenShift documentation](https://docs.redhat.com/en/documentation/openshift_container_platform/4.18/html/monitoring/configuring-core-platform-monitoring#configuring-remote-write-storage_configuring-metrics) the following manifest needs to be applied to send platform metrics to VictoriaMetrics:

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: cluster-monitoring-config
  namespace: openshift-monitoring
data:
  config.yaml: |
    prometheusK8s:
      remoteWrite:

>       - url: "http://vminsert-hub.victoriametrics.svc.cluster.local:8480/insert/1/prometheus/api/v1/write"

Reviewer: It would be nice to mention that this URL could be a single-node URL and that it could contain tenant. Similarly how vmagent integration describes it https://docs.victoriametrics.com/victoriametrics/data-ingestion/vmagent/

        writeRelabelConfigs:
        - regex: 'prometheus_replica'
          action: labeldrop
```
This instructs Prometheus to copy collected metrics to specified VictoriaMetrics ingester.

> Note, that OpenShift uses two Prometheus replicas for HA configuration. Metric series are identified by `prometheus_replica` label. This is excessive on VictoriaMetrics side, so this label needs to be dropped to avoid inserting duplicate metrics.

Author: Yeah, that refers to
```
        writeRelabelConfigs:
        - regex: 'prometheus_replica'
          action: labeldrop
```

Initially I thought this part should be optional, but it's best to have it included in the default config we're suggesting. Not quite sure how to arrange this idea in the guide - any suggestions?

> Note, that OpenShift uses two Prometheus replicas for HA configuration. Metric series are identified by `prometheus_replica` label. This is excessive on VictoriaMetrics side, so this label needs to be dropped to avoid inserting duplicate metrics.

Reviewer 2: I'd mention that there is an example, a few lines of text below.

> Note, that OpenShift uses two Prometheus replicas for HA configuration. Metric series are identified by `prometheus_replica` label. This is excessive on VictoriaMetrics side, so this label needs to be dropped to avoid inserting duplicate metrics.

Reviewer: ```suggestion
Note, that OpenShift uses two Prometheus replicas for HA configuration. Each replica adds `prometheus_replica` label to the forwarded metrics. This is excessive on VictoriaMetrics side, so this label needs to be dropped to avoid inserting duplicate metrics.
```
